src/main/kotlin/com/expedia/graphql/schema/schemaAnnotations.kt

```diff
@@ -68,17 +86,35 @@ internal fun KAnnotatedElement.directives() =
                 annotation.annotationClass.simpleName ?: throw CouldNotGetNameOfAnnotationException(annotation.annotationClass)
             }
             builder.name(CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_CAMEL, name))
+                .validLocations(*directiveInfo.locations)
+                .description(directiveInfo.description)
 
-            annotation::class.properties().forEach { prop ->
-                val propertyName = prop.name
-                val value = prop.call(annotation)
-                @Suppress("Detekt.UnsafeCast")
-                val type = graphQLScalar(prop.returnType) as GraphQLInputType
-                builder.argument(GraphQLArgument.newArgument().name(propertyName).value(value).type(type).build())
+            when {
+                (this is KClass<*>) && this.javaObjectType.isAnnotation -> this.memberProperties.forEach(buildArgument(builder))
+                else -> annotation::class.properties().forEach(buildArgument(annotation, builder))
             }
 
-            builder.build()
-        }
+        builder.build()
+    }
+}
+
+private fun buildArgument(annotation: Annotation, builder: GraphQLDirective.Builder): (KFunction<*>) -> Unit {
```

Comment:
can you rename it to something more meaningful instead of overloading it for both function and property?